### DISCUSSION TITLE
Few heading questions

### DISCUSSION THREAD

**Discussion by Inerska** [statiqdev/Discussions#73] [Q&A] [answered]
Hello, just getting started with Statiq, what is really cool.

I'm just using CleanBlog (https://github.com/statiqdev/CleanBlog) as a theme, but when I want to test it `dotnet run -- preview`, it doesn't load.cshtml files in output folder ?
Where are the imports for razor pages?

Do I have to write the Statiq credit on my footer page ? 

How to deploy in Netlify ? If I'm .gitignoring the app config file, how GitHub will know my API key to deploy it on Netlify?
How to create articles and rendering theme ?

Thanks in advance.
Alexis

**Comment by daveaglick** [comment]
> Where are the imports for razor pages?

When using CleanBlog all `.cshtml` files should be automatically processed and rendered - Statiq handles Razor, Markdown, and Handlebars files automatically. Are you able to paste the console output so I can take a look?

> Do I have to write the Statiq credit on my footer page?

Nope, not at all. No attribution is necessary.

> How to deploy in Netlify?

There's a few ways to handle Netlify deployments. If building inside Netlify CI, I think it'll pick up on certain patterns (I've never done it, but I know others have been successful). You can also easily deploy to Netlify from any CI server or even locally using Statiq directly with the `deploy` command (I.e. `dotnet run -- deploy`): https://statiq.dev/web/deployment/netlify

> If I'm .gitignoring the app config file, how GitHub will know my API key to deploy it on Netlify?

Statiq will combine the settings in your config files with all the currently defined environment variables. In GitHub Actions, you can define additional secrets and those will be injected into your CI process as environment variables: https://docs.github.com/en/actions/reference/encrypted-secrets

> How to create articles and rendering theme ?

In CleanBlog, posts can be placed in the `posts` folder. You can see an example in my own blog (which uses CleanBlog) here: https://github.com/daveaglick/daveaglick/tree/develop/input/posts

Please don't hesitate to ask if any of this isn't totally clear or you have additional questions!

**Comment by Inerska** [comment, marked as answer]
> 
> 
> > Where are the imports for razor pages?
> 
> When using CleanBlog all `.cshtml` files should be automatically processed and rendered - Statiq handles Razor, Markdown, and Handlebars files automatically. Are you able to paste the console output so I can take a look?
> 
> > Do I have to write the Statiq credit on my footer page?
> 
> Nope, not at all. No attribution is necessary.
> 
> > How to deploy in Netlify?
> 
> There's a few ways to handle Netlify deployments. If building inside Netlify CI, I think it'll pick up on certain patterns (I've never done it, but I know others have been successful). You can also easily deploy to Netlify from any CI server or even locally using Statiq directly with the `deploy` command (I.e. `dotnet run -- deploy`): https://statiq.dev/web/deployment/netlify
> 
> > If I'm .gitignoring the app config file, how GitHub will know my API key to deploy it on Netlify?
> 
> Statiq will combine the settings in your config files with all the currently defined environment variables. In GitHub Actions, you can define additional secrets and those will be injected into your CI process as environment variables: https://docs.github.com/en/actions/reference/encrypted-secrets
> 
> > How to create articles and rendering theme ?
> 
> In CleanBlog, posts can be placed in the `posts` folder. You can see an example in my own blog (which uses CleanBlog) here: https://github.com/daveaglick/daveaglick/tree/develop/input/posts
> 
> Please don't hesitate to ask if any of this isn't totally clear or you have additional questions!


That is, the console output from `cd` to the directory path to the `dotnet run -- preview`
https://pastebin.com/bZsHPN3q

And here's the content of my output folder after the `dotnet run -- preview`
![image](https://user-images.githubusercontent.com/50674490/121090071-4fcb5d80-c7e8-11eb-8447-1a637408709e.png)

And here's the project file structure
![image](https://user-images.githubusercontent.com/50674490/121090489-f6176300-c7e8-11eb-8b09-a82c5b55b59e.png)

**Reply by daveaglick** [reply]
Thanks, that's helpful. I don't actually see any `.cshtml` files in your `input` folder, am I missing something?

**Reply by Inerska** [reply]
> 
> 
> Thanks, that's helpful. I don't actually see any `.cshtml` files in your `input` folder, am I missing something?

All `.cshtml` files are in my `theme/input` (CleanBlog) folder

**Reply by daveaglick** [reply]
Ah, got it. So you're wondering why the files like `index.cshtml` from the theme aren't being rendered, is that right? The key here is that the three `.cshtml` files in the CleanBlog theme are all archives: https://statiq.dev/web/content-and-data/archives. That is, they're generated based on the posts in the site. You can see several `Archive...` values in their front-matter if you open them in a text editor. If there aren't any posts to drive the archives, nothing gets output.

If you want a "plain" index page without the archive support, you can add an `index.cshtml` to your own `input` folder and that will override the one in the theme (alternatively you can replace the one in the theme directly). You can also test out normal Razor files by adding something like `test.cshtml` to either your `input` folder or the `theme/input` folder and see it get rendered.

I think this answers your question - but let me know if not!

**Reply by Inerska** [reply]
Oh, I see! So I just have to create a post to make it all work, is that right ? 
Thanks a lot, I was struggling so hard haha

**Reply by daveaglick** [reply]
Yep! And that's actually great feedback - I can see where it's really confusing out of the box. I'll add a note to the CleanBlog readme about this behavior.
